packages/suite-desktop-core/src/modules/bioAuthModule.ts

```diff
@@ -3,189 +3,267 @@ import { systemPreferences } from 'electron';
+    mainWindowProxy,
+    store,
+}: {
+    mainWindowProxy: Dependencies['mainWindowProxy'];
+    store: Dependencies['store'];
+}) => {
+    let bioAuth: BioAuth | undefined;
+
+    const onLoad = () => {
+        if (bioAuth) return;
+        const { logger } = global;
+        logger.info('bioAuth', `Loading. Enabled in store: ${store.getBioAuthSettings().enabled}`);
+
+        const constructorParams = {
+            logger,
+            enabled: store.getBioAuthSettings().enabled || false,
         };
-    });
+        if (isMacOs()) {
+            bioAuth = new BioAuthMac(constructorParams);
+            logger.info('bioAuth', 'Loaded for Mac');
+        } else if (isWindows()) {
+            bioAuth = new BioAuthWindows(constructorParams);
+            logger.info('bioAuth', 'Loaded for Windows');
+        } else if (isLinux()) {
+            bioAuth = new BioAuthLinux(constructorParams);
+            logger.info('bioAuth', 'Loaded for Linux');
+        } else {
+            logger.info('bioAuth', 'Unsupported platform for BioAuth');
 
-    ipcMain.handle('bio-auth/is-bio-auth-available', ipcEvent => {
-        validateIpcMessage({ ipcEvent });
+            return;
+        }
 
-        return false;
-    });
-};
+        ipcMain.handle('bio-auth/is-bio-auth-available', ipcEvent => {
+            validateIpcMessage({ ipcEvent });
 
-export const initBioAuthModule = () => {
-    let loaded = false;
-    let destroy: null | (() => void) = null;
+            if (!bioAuth) throw new Error('BioAuth module is not initialized');
 
-    const onLoad = async () => {
-        if (loaded) return;
-        const { logger } = global;
-        logger.info('bioAuth', 'Loading');
+            return bioAuth.isAvailable();
+        });
 
-        if (isMacOs()) {
-            loaded = true;
-            loadMac();
-            logger.info('bioAuth', 'Loaded for Mac');
-        }
+        ipcMain.handle('bio-auth/set-bio-auth-settings', (ipcEvent, params) => {
+            validateIpcMessage({ ipcEvent });
 
-        if (isWindows()) {
-            loaded = true;
-            const { destroy: destroyWin } = await loadWin();
-            destroy = destroyWin;
-            logger.info('bioAuth', 'Loaded for Windows');
-        }
+            store.setBioAuthSettings(params);
+        });
 
-        if (isLinux()) {
-            loaded = true;
-            loadLinux();
-        }
+        ipcMain.handle('bio-auth/get-bio-auth-settings', ipcEvent => {
+            validateIpcMessage({ ipcEvent });
+
+            return { enabled: false, ...store.getBioAuthSettings() };
+        });
+
+        store.onBioAuthSettingsChange(newSettings => {
+            if (!newSettings || typeof newSettings.enabled !== 'boolean') return;
+            mainWindowProxy
+                .getInstance()
+                // @ts-expect-error ...
+                ?.webContents.send('bio-auth/settings-changed', newSettings);
+            bioAuth?.setEnabled(newSettings.enabled);
```

Comment:
More verbose but TS works 🙂 
```suggestion
            if (!newSettings) return;
            const { enabled } = newSettings;
            if (typeof enabled !== 'boolean') return;

            mainWindowProxy
                .getInstance()
                ?.webContents.send('bio-auth/settings-changed', { enabled });
            bioAuth?.setEnabled(enabled);
```

Reply:
f6783e985816ca218749ab622948eba909f3cd96